Libpng 1.6.28

catalinr opened the pull request:
Not sure if `PNG_ZLIB_VERNUM` in `src/png/pnglibconf.h` should be kept as 0 or the real zlib value.

catalinr committed:
Upgrade bundled libpng to 1.6.28.

catalinr committed:
Add libpng upgrade example for MSW.

vadz commented:
Thanks for doing this! Would you already have a list of changes since 1.6.21 affecting the use of the library in wxWidgets applications (e.g. not related to build system) to add to the commit message/our own changelog by chance?

catalinr commented:
The complete list of changes is included with libpng, in `src/png/CHANGES`.
Here is a list of changes that could be considered more interesting:
- Quiet API change that limits in-memory image size (uncompressed) to less than 4GByte and image row size (stride) to less than 2GByte.
- Corrected error in PNG_IMAGE_PNG_SIZE_MAX that underreported the palette size.
- Stop a potential memory leak in png_set_tRNS().
- Fixed the progressive reader to handle empty first IDAT chunk properly.
- Avoid potential overflow of the PNG_IMAGE_SIZE macro.
- Correct filter heuristic overflow handling.
- Fixed the case where PNG_IMAGE_BUFFER_SIZE can overflow in the application as a result of the application using an increased 'row_stride'.
- Optimized absolute value calculation in filter selection.
- Avoid filter-selection heuristic sum calculations in cases where only one filter is a candidate for selection. This trades off code size for speed.
- Reject oversized iCCP profile immediately.
- Fixed handling zero length IDAT in pngfix.
- Do not issue a png_error() on read in png_set_pCAL() because png_handle_pCAL has allocated memory that libpng needs to free.
- Several changes for ADLER32.
- Use zlib-1.2.8.1 inflateValidate() instead of inflateReset2() to optionally avoid ADLER32 evaluation.
- Fixed a potential null pointer dereference in png_set_text_2().
- Added INTEL-SSE2 support.
- Added MIPS support.
- Speed improvements for INTEL-SSE.


p.s. `PNG_ZLIB_VERNUM` is still set to non-0 value.

vadz commented:
I did see the full change log, but I don't know how to condense it to something useful... I'll probably just say "refer to libpng changelog".

It looks like `PNG_ZLIB_VERNUM` is related to the Appveyor build failure, have you looked at this one?

catalinr committed:
Define `PNG_ZLIB_VERNUM` as 0.
